protostar/configuration_file/configuration_file_facade_v2.py

```diff
@@ -49,9 +49,9 @@ def from_v1(cls, v1: ConfigurationFileV1Model, protostar_version: str) -> Self:
         )
 
 
-class ConfigurationFileV2(
```

Comment:
> word facade is too vague imho and does not say what this class is doing except saying that it follows a book about design patterns

Pattern names convey information about the intention.

The intention of Facade is providing a simplified interface, and this is what this class does. `ConfigurationFileFacade` could be understood as `SimplifiedInterfaceToConfigurationFile`.

![Screen Shot 2022-09-22 at 16 20 43](https://user-images.githubusercontent.com/9629577/191772512-78c21856-aa9b-4ba4-bd57-1f4755a8816b.png)

The name `ConfigurationFileV2IO` is misleading because this class neither reads nor writes the configuration file.

Reply:
For me the simplified interface is the model that is renamed in this PR.

Reply:
> The simplified interface is the model that is renamed in this PR as far as I understand. 

The "model" in this case  is a data structure that represents the raw configuration file in Python. It doesn't provide the simplified interface. The facade parses, prepares paths and makes retrieving data easier for the caller.

> Please reflect what this class does in its name.

I couldn't find a better name, but I think it's better than `IO` suffix. IMO, the current name conveys the intention pretty well.

Reply:
> The facade parses, prepares paths and makes retrieving data easier for the caller.

Editor? Accessor?

Reply:
Alternatively, I can use more vague name like Manager, Processor etc.

Reply:
Yes, that's why I'm looking for more specific name. Something that refers to "text editor" metaphor

Reply:
@MaksymilianDemitraszek